include/kitty/print.hpp

```diff
@@ -478,4 +481,93 @@ std::string anf_to_expression( const ternary_truth_table<TT>& anf )
   return anf_to_expression( anf._bits );
 }
 
-} /* namespace kitty */
\ No newline at end of file
+/*! \brief Creates an expression in the sum of products format.
+
+  Does not perform any optimizations. Useful when constructing GENLIB-compatible expressions
+
+  \param tt Truth table
+  \param os Output stream
+ */
+template<typename TT, typename = std::enable_if_t<is_completely_specified_truth_table<TT>::value>>
+void print_sop_expression(TT tt, std::ostream& os = std::cout )
+{
+  /* compare to constants */
+  if ( is_const0( tt ) )
+  {
+    os << "CONST0";
+    return;
+  }
+  else if ( is_const0( ~tt ) )
+  {
+    os << "CONST1";
+    return;
+  }
+
+  /* extract product terms */ 
+  auto cubes = kitty::isop(tt);
+
+  bool first_cube = true; // Controls insertion of '|'. It's false for the first cube to avoid leading '|'.
+
+  
+  /* write product terms */ 
+  for (auto cube : cubes)
+  {
+    auto bits = cube._bits;
+    auto mask = cube._mask;
+
+    bool brackets = (__builtin_popcount(mask) > 1);
+
+    if (!first_cube)
```

Comment:
In the formatting for this library we expect spaces after opening and before closing parentheses.